### DISCUSSION TITLE
How to check NextJS types before pushing

### DISCUSSION THREAD

**Discussion by frankdilo** [vercel/next.js#45181] [Help]
Hi 👋

I am trying to reduce CI failures due to TypeScript errors in our NextJS projects, and I have trouble finding the right approach.

I tried adding a pre-push hook that runs a `next build`, but that's too slow, and we decided to turn it off. 

Ideally, I would like only to check types (and not do a full build) as that's 80% of errors that break our CI, and checking only those would be faster than a full NextJS build. Checking types with `tsc` is inconsistent with what `next build` does. For example, in our project, `next build` succeeds, and `tsc --noEmit --incremental --project tsconfig.json` fails with minor errors that we don't want to bother with.

Any solution?

**Comment by icyJoseph** [comment]
Wait, so, next build with TypeScript passes? but then you do extra checks on your CI, and those fail? It sounds like it's not the same TypeScript version that's being used in both processes? 

It's like sometimes people have TS a.b.c in VSCode, and, by coincidence, so does their Next.js project, but as soon as VSCode updates, and it uses version x.y.z, then they start to see code break during build time, but not during development. In that case one has to tell VSCode to use the workspace TS version.

Are you sure both projects are using the same TypeScript versions?

**Reply by frankdilo** [reply]
How do I check the TypeScript version that's being used by Next?

**Reply by icyJoseph** [reply]
It should be listed in the `package.json` of the project. One could try `npm explain typescript`, or `yarn why typescript`.
